### DISCUSSION TITLE
Oxy Color Palette Colors and Legend do not match

### DISCUSSION THREAD

**Discussion by aad5232** [oxyplot/oxyplot#1908] [Q&A] [answered]
Hello, so recently I was working on generating Bubble chart using Oxyplot, and I created my own palette of colors. I added those colors to a new linear color axis and added to my Plot model's axes. But the colors of the bubble chart do not match its legend. Before adding the linear color axis the colors were matching. How do I sync or match the colors of my palette with the legend? 
![The bubble chart](https://user-images.githubusercontent.com/90162079/183528273-d9572832-f1d4-4561-ae02-cfb420e0d706.png)

**Comment by VisualMelon** [comment, marked as answer]
> How do I sync or match the colors of my palette with the legend?

The legend color cannot be changed automatically. The only way to visually identify the series that I can think of when you have a color axis is to give them a specific outline. (i.e. set `MarkerStroke` and `MarkerFill` explicitly per-series and use a non-zero `MarkerStrokeThickness`).

If you have better ideas for functionality to help with bubble plots, feel free to share them. This isn't a usage that has had much consideration, so I'm afraid the APIs are probably not well aligned with your intentions. It may make sense to provide a dedicated bubble plot with a simpler (bubble specific) API.

<hr />

Word of warning: if you are using the marker size to show the value and using a 'solid' marker, then consider scaling the marker size with the square-root of the value (assuming you are not already), as people tend to instinctively compare the areas of things rather than the widths.

**Reply by aad5232** [reply]
So, by specifying just the MarkerFill colors for each series to be the same as the bubble colors for that series, it helped in matching the legend color to the bubble color. Regarding functionality to help with bubble plots, I am creating a dictionary of series and their respective colors and storing those colors by converting them to uint from oxycolor which I am converting to oxycolor from ARGB values from my color palette. Since to create bubbles or scatterpoints need color value in int, and then using the same colors per series from dictionary while creating a series for MarkerFill as well as for scatterpoints per scatterseries.
